### PR TITLE
Jwt authentication

### PR DESCRIPTION
**Pull request by diegoportela99**
Authentication for users

### CODE REVIEW

File: HELPS/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        [AllowAnonymous]
+        [HttpPost("authenticate")]
+        public IActionResult Authenticate([FromBody]User userParam)
+        {
+            var user = _userService.Authenticate(userParam.Username, userParam.Password);
+
+            if (user == null)
+                return BadRequest(new { message = "Username or password is incorrect" });
```
Comment (ghost) [inline HELPS/Controllers/UserController.cs]: Wrong response code, should be 401

File: HELPS/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+            return Ok(user);
```
Comment (ghost) [inline HELPS/Controllers/UserController.cs]: Returns invalid object (see spec), Token should be renamed to AccessToken and should include isAdmin

File: HELPS/Startup.cs [hunk 3]
```diff
@@ -34,7 +39,35 @@ public void ConfigureServices(IServiceCollection services)
...
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = new SymmetricSecurityKey(key),
+                    ValidateIssuer = false,
+                    ValidateAudience = false
+                };
+            });
+
+            // configure DI for application services
+            services.AddScoped<IUserService, UserService>();
+
```
Comment (ghost) [inline HELPS/Startup.cs:71]: Missing UseAuthentication()?

File: HELPS/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        private IUserService _userService;
+
+        public UsersController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("authenticate")]
+        public IActionResult Authenticate([FromBody]User userParam)
```
Comment (ghost) [inline HELPS/Controllers/UserController.cs:23]: This endpoint does not match spec, should be api/login

File: HELPS/Controllers/UserController.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+            if (user == null)
+                return BadRequest(new { message = "Username or password is incorrect" });
+
+            return Ok(user);
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
```
Comment (ghost) [inline HELPS/Controllers/UserController.cs:38]: I am assuming this is for testing but definitely should not be included in production

**Review by ghost — CHANGES_REQUESTED** [review]
I did some very surface level testing and found a few issues so take this review with a grain of salt. 

I would suggest fixing the endpoints and protecting the user controller before merging (definitely before merging into develop) but you can merge this as is if you would prefer.

File: HELPS/Startup.cs [hunk 6]
```diff
@@ -66,7 +66,8 @@ public void ConfigureServices(IServiceCollection services)
             });
 
             // configure DI for application services
-            services.AddScoped<IUserService, UserService>();
+            services.AddScoped<IUserService, UserService>()
+                    .AddAuthentication();
```
Comment (diegoportela99) [inline HELPS/Startup.cs:70]: Is this all that's needed? i'm not too sure

**Pull request merged**